x/sync/manager.go

```diff
@@ -604,20 +604,28 @@ func (m *Manager) completeWorkItem(ctx context.Context, work *workItem, largestH
 		}
 	}
 
-	// completed the range [work.start, lastKey], log and record in the completed work heap
-	m.config.Log.Info("completed range",
-		zap.Binary("start", work.start),
-		zap.Binary("end", largestHandledKey),
-	)
-	if m.getTargetRoot() == rootID {
+	// Process [work] while holding [syncTargetLock] to ensure that object
+	// is added to the right queue, even if a target update is triggered
+	m.syncTargetLock.RLock()
+	var stale bool
```

Comment:
small nit - can we define `stale` based on the comparison rather than the control flow?
```suggestion
	stale := m.config.TargetRoot != rootID
```